File: src/main/kotlin/com/scottscmo/ui/OutputDisplay.java
package com.scottscmo.ui;

import com.scottscmo.Application;
import javax.swing.JOptionPane;

public final class OutputDisplay {

    public static void show(String msg) {
        JOptionPane.showMessageDialog(Application.get(), msg);
    }

    public static void error(String msg, Exception e) {
        JOptionPane.showMessageDialog(Application.get(), msg, msg, JOptionPane.ERROR_MESSAGE);
        Application.getLogger().error(msg, e);
    }
}

File: src/main/kotlin/com/scottscmo/ui/OutputDisplay.kt
package com.scottscmo.ui

import com.scottscmo.Application
import java.lang.Exception
import javax.swing.JOptionPane

object OutputDisplay {

    fun show(msg: String?) {
        JOptionPane.showMessageDialog(Application.get(), msg)
    }

    fun error(msg: String?) {
        JOptionPane.showMessageDialog(Application.get(), msg, msg, JOptionPane.ERROR_MESSAGE)
    }

    fun error(e: Exception) {
        error(e.message + "\n" + e.stackTrace.copyOfRange(0, 10).joinToString("\n"))
    }
}
